common/dynamicconfig/constants.go

```diff
@@ -888,6 +888,16 @@ used when the first cache layer has a miss. Requires server restart for change t
 		32,
 		`MaxCallbacksPerWorkflow is the maximum number of callbacks that can be attached to a workflow.`,
 	)
+	FrontendLinkMaxSize = NewNamespaceIntSetting(
+		"frontend.linkMaxSize",
+		4000, // Links may include a workflow ID and namespace name, both of which are limited to a length of 1000.
```

Comment:
They could also have a run ID but I would hope 4000 is still enough room.

Reply:
Yeah, that's why I gave some extra padding on top of the workflow ID + namespace.